src/pages/Reviews.js

```diff
@@ -1,42 +1,20 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
 import axios from 'axios';
 import { useUserContext } from "../context/user_context";
 
-const Reviews = ({ productId }) => {
+const Reviews = ({ productId, reviews, onReviewChange }) => {
   // Fetch current user from context provider, and store it in the `user` variable
   const { currentUser: user } = useUserContext();
   //console.log(user.userId); //user id
 
   const [message, setMessage] = useState('');
-  const [reviews, setReviews] = useState([]);
   const [formData, setFormData] = useState({
-    rating: Number,
+    rating: '',
     title: '',
     comment: '',
     product: productId
   });
 
-  // Get all reviews
-  useEffect(() => {
-    getAllReviews(productId);
-  }, [productId]);
-
-  const getAllReviews = async (productId) => {
-    try {
-      const response = await axios.get(
-        'https://ecommerce-6kwa.onrender.com/api/v1/reviews',
-        { withCredentials: true }
-      );
-      const allReviews = response.data.reviews;
-      const filteredReviews = allReviews.filter((review)=> review?.product?._id === productId)
-
-      setReviews(filteredReviews);
-      console.log('this is the result', filteredReviews);
-    } catch (error) {
-      console.error(error);
-    }
-  };
```

Comment:
This was fine; but instead of fetching all the reviews from the backend and then making a filtering in our frontend, we can just use the list of `reviews` that are embedded in the single product response anyway. This works because your backend code defined a virtual field called `reviews`: https://github.com/NatachaKey/ecommerce/blob/main/models/Product.js#L68